File: frontend/ClickAndDrive/src/app/services/passenger.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { EmptyError, Observable } from 'rxjs';
import { RouteFromFavorites } from './models/route-from-favorites';
import { PassengerRideHistory } from './models/passenger-ride-history';

@Injectable({
    providedIn: 'root'
})
export class PassengerService {
    private apiUrl = 'http://localhost:8080/api/passenger';

    constructor(
        private http: HttpClient
    ) {}

    // GET users favorite routes
    getFavoriteRoutes(passengerId: number): Observable<RouteFromFavorites[]> {
        return this.http.get<RouteFromFavorites[]>(`${this.apiUrl}/${passengerId}/favorite-routes`);
    }

    // Remove a route from favorites
    removeFavoriteRoute(passengerId: number, routeId: number): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${passengerId}/${routeId}/remove-route`);
    }

    getPassengerHistory(passengerId: number) {
        return this.http.get<PassengerRideHistory[]>(
            `http://localhost:8080/api/passenger/${passengerId}/ride-history`
        );
    }

    getRideDetails(rideId: number) {
        return this.http.get<any>(`http://localhost:8080/api/passenger/rides/${rideId}/details`);
    }
}

File: mobile-version/Navbar-shortened/app/src/main/java/com/example/clickanddrive/clients/services/PassengerService.java
package com.example.clickanddrive.clients.services;

import com.example.clickanddrive.dtosample.responses.RouteFromFavoritesResponse;
import com.example.clickanddrive.dtosample.responses.PassengerRideDetailsResponse;
import com.example.clickanddrive.dtosample.responses.PassengerRideHistoryResponse;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface PassengerService {

    // GET for retrieving passengers favorite routes
    @GET("passenger/{passengerId}/favorite-routes")
    Call<List<RouteFromFavoritesResponse>> getFavoriteRoutes(@Path("passengerId") Long passengerId);

    // Remove a route from favorites list
    @DELETE("passenger/{passengerId}/{routeId}/remove-route")
    Call<Void> removeFromFavorites(@Path("passengerId") Long passengerId, @Path("routeId") Long routeId);

    @GET("passenger/{id}/ride-history")
    Call<List<PassengerRideHistoryResponse>> getPassengerRideHistory(@Path("id") Long id);

    @GET("passenger/rides/{rideId}/details")
    Call<PassengerRideDetailsResponse> getRideDetails(@Path("rideId") Long rideId);
}
